pyialarm/pyialarm.py

```diff
@@ -1,218 +1,393 @@
+            raise ConnectionError(error_message)
+
+        result = []
+        for i, status in enumerate(zone_status):
+            zone_id = i + 1
+
+            status_list = []
+
+            if status & StatusType.ZONE_IN_USE:
+                status_list.append(StatusType.ZONE_IN_USE)
+            if status & StatusType.ZONE_ALARM:
+                status_list.append(StatusType.ZONE_ALARM)
+            if status & StatusType.ZONE_BYPASS:
+                status_list.append(StatusType.ZONE_BYPASS)
+            if status & StatusType.ZONE_FAULT:
+                status_list.append(StatusType.ZONE_FAULT)
+            if status & StatusType.ZONE_LOW_BATTERY:
+                status_list.append(StatusType.ZONE_LOW_BATTERY)
+            if status & StatusType.ZONE_LOSS:
+                status_list.append(StatusType.ZONE_LOSS)
+
+            if not status_list:
+                status_list.append(StatusType.ZONE_NOT_USED)
+
+            zone_name = zone_name_map.get(zone_id, "Unknown")
+
+            zone_item: ZoneStatusType = {
+                "zone_id": zone_id,
+                "name": zone_name,
+                "types": status_list,
+            }
+            result.append(zone_item)
+
+        return result
+
+    async def get_status(self) -> int:
+        command: OrderedDict[str, Optional[Any]] = OrderedDict()
+        command["DevStatus"] = None
+        command["Err"] = None
+
+        alarm_status: dict[str, Any] = await self._send_request(
+            "/Root/Host/GetAlarmStatus", command
+        )
+
         if alarm_status is None:
-            raise ConnectionError('An error occurred trying to connect to the alarm '
-                                  'system')
+            error_message = "An error occurred trying to connect to the alarm system"
+            raise ConnectionError(error_message)
 
         status = int(alarm_status.get("DevStatus", -1))
         if status == -1:
-            raise ConnectionError('Received an unexpected reply from the alarm')
-
-        if not include_memory_feature:
-            return status
-
-        zone_alarm = False
-        command = OrderedDict()
-        command['Total'] = None
-        command['Offset'] = 'S32,0,0|0'
-        command['Ln'] = None
-        command['Err'] = None
-        zone_status = self._send_request_list('/Root/Host/GetByWay', command)
-        self._close_connection()
-        if zone_status is None:
-            raise ConnectionError('An error occurred trying to connect to the alarm '
-                                  'system')
-        for zone in zone_status:
-            if zone & self.ZONE_ALARM:
-                zone_alarm = True
+            error_message = "Received an unexpected reply from the alarm"
+            raise ConnectionError(error_message)
+
+        if status in {self.ARMED_AWAY, self.ARMED_STAY}:
+            zone_status: list[ZoneStatusType] = await self.get_zone_status()
```

Comment:
Here you are sending a `GetAlarmStatus` request and then a `GetByWay` back to back. When I initially attempted it I noticed that it made the alarm would eventually stop responding to the `GetbyWay`. That why in https://github.com/RyuzakiKK/pyialarm/commit/2764bbdb0e06a4d0d231eab10a7cbbee038565ad I was using a one second timeout.

If it works, great! If it doesn't, then we probably need to add some wait before sending the second request.